app/utility/observer.py

```diff
@@ -0,0 +1,18 @@
+class Observer():
+    _observers = []
+
+    def __init__(self):
+        self._observers.append(self)
+        self._observed_events = []
+
+    def observe(self, event_name, callback_fn):
+        self._observed_events.append({'name': event_name, 'callback': callback_fn})
+
+
+class Event():
```

Comment:
let's rename this file to 'events'. "event-driven" is more universally understood